linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/CoreJSLib.scala

```diff
@@ -731,278 +694,244 @@ private[emitter] object CoreJSLib {
             StringArgConstructorName, str("/ by zero")))
       }
 
-      locally {
-        defineFunction("intDiv", paramList(x, y), {
+      def wrapBigInt64(tree: Tree): Tree =
+        Apply(genIdentBracketSelect(BigIntRef, "asIntN"), 64 :: tree :: Nil)
+
+      Block(
+        defineFunction2("intDiv") { (x, y) =>
           If(y === 0, throwDivByZero, {
             Return((x / y) | 0)
           })
-        })
-        defineFunction("intMod", paramList(x, y), {
+        },
+
+        defineFunction2("intMod") { (x, y) =>
           If(y === 0, throwDivByZero, {
             Return((x % y) | 0)
           })
-        })
-      }
+        },
 
-      locally {
-        defineFunction("doubleToInt", paramList(x), {
+        defineFunction1("doubleToInt") { x =>
           Return(If(x > 2147483647, 2147483647, If(x < -2147483648, -2147483648, x | 0)))
-        })
-      }
-
-      if (allowBigIntsForLongs) {
-        def wrapBigInt64(tree: Tree): Tree =
-          Apply(genIdentBracketSelect(BigIntRef, "asIntN"), 64 :: tree :: Nil)
+        },
 
-        defineFunction("longDiv", paramList(x, y), {
-          If(y === BigIntLiteral(0), throwDivByZero, {
-            Return(wrapBigInt64(x / y))
-          })
-        })
-        defineFunction("longMod", paramList(x, y), {
-          If(y === BigIntLiteral(0), throwDivByZero, {
-            Return(wrapBigInt64(x % y))
-          })
-        })
+        condBlock(allowBigIntsForLongs)(
+          defineFunction2("longDiv") { (x, y) =>
+            If(y === BigIntLiteral(0), throwDivByZero, {
+              Return(wrapBigInt64(x / y))
+            })
+          },
+          defineFunction2("longMod") { (x, y) =>
+            If(y === BigIntLiteral(0), throwDivByZero, {
+              Return(wrapBigInt64(x % y))
+            })
+          },
 
-        defineFunction("doubleToLong", paramList(x), Return {
-          If(x < double(-9223372036854775808.0), { // -2^63
-            BigIntLiteral(-9223372036854775808L)
-          }, {
-            If (x >= double(9223372036854775808.0), { // 2^63
-              BigIntLiteral(9223372036854775807L)
+          defineFunction1("doubleToLong") { x => Return {
+            If(x < double(-9223372036854775808.0), { // -2^63
+              BigIntLiteral(-9223372036854775808L)
             }, {
-              If (x !== x, { // NaN
-                BigIntLiteral(0L)
+              If (x >= double(9223372036854775808.0), { // 2^63
+                BigIntLiteral(9223372036854775807L)
               }, {
-                Apply(BigIntRef,
-                    Apply(genIdentBracketSelect(MathRef, "trunc"), x :: Nil) :: Nil)
+                If (x !== x, { // NaN
+                  BigIntLiteral(0L)
+                }, {
+                  Apply(BigIntRef,
+                      Apply(genIdentBracketSelect(MathRef, "trunc"), x :: Nil) :: Nil)
+                })
               })
             })
-          })
-        })
-      }
+          }}
```

Comment:
Hum, there's no precedent for for having a double `}}`, ever. I'm not even sure we can have a closing brace anywhere but at the start of a line (except in `import`s). I see that this comes from
```scala
defineFunction1("doubleToLong") { x => Return {
```
but I don't think we should allow this. A lambda formatted with `{ params =>` should always have its body start on the next line.

IMO there are two solutions here. Either put `Return {` on the next line, and indent everything one level more. Or format it as
```scala
defineFunction1("doubleToLong")(x => Return {
  ...
})
```